scale/docs/rest/v6/strike.rst

```diff
@@ -0,0 +1,496 @@
+| name               | String            | The identifying name of the Strike process used for queries.                   |
++--------------------+-------------------+--------------------------------------------------------------------------------+
+| title              | String            | The human readable display name of the Strike process.                         |
++--------------------+-------------------+--------------------------------------------------------------------------------+
+| description        | String            | A longer description of the Strike process.                                    |
++--------------------+-------------------+--------------------------------------------------------------------------------+
+| job                | JSON Object       | The job that is associated with the Strike process.                            |
+|                    |                   | (See :ref:`Job Details <rest_job_details>`)                                    |
++--------------------+-------------------+--------------------------------------------------------------------------------+
+| created            | ISO-8601 Datetime | When the associated database model was initially created.                      |
++--------------------+-------------------+--------------------------------------------------------------------------------+
+| last_modified      | ISO-8601 Datetime | When the associated database model was last saved.                             |
++--------------------+-------------------+--------------------------------------------------------------------------------+
+| configuration      | JSON Object       | JSON defining the Strike configuration.                                        |
+|                    |                   | (See :ref:`architecture_strike_spec`)                                          |
++--------------------+-------------------+--------------------------------------------------------------------------------+
+
+.. _rest_v6_strike_validate:
+
+**Example POST /v6/strikes/validation/ API call**
+
+Request: POST http://.../v6/strikes/validation/
+
+.. code-block:: javascript 
+
+    { 
+        "name": "my-strike-process", 
+        "title": "My Strike Process", 
+        "description": "This is my Strike process for detecting my favorite files!", 
+        "configuration": { 
+            "version": "2.0", 
+            "workspace": "my-workspace", 
+            "monitor": { 
+                "type": "dir-watcher", 
+                "transfer_suffix": "_tmp" 
+            }, 
+            "files_to_ingest": [{ 
+                "filename_regex": ".*txt" 
+            }] 
+        } 
+    } 
+
+Response: 200 OK
+
+.. code-block:: javascript 
+ 
+    { 
+        "warnings": [ 
+            "id": "mount_change", 
+            "details": "Changing the mount path may disrupt file monitoring." 
+        ] 
```

Comment:
The return for this validation should mirror batch validation. That is the standard way we'll do validation API calls.

Reply:
Check out the validation warning and error classes in util.validation and see how batch uses them. Shouldn't be too big of a change.